### ISSUE TITLE
AbortError on Firefox

### ISSUE THREAD

**Issue by jandede** [andy-portmen/native-client#123]
Version: 111.0.1 Mozilla Firefox Snap for Ubuntu
Ubuntu 23.04

I installed the script with ./install.sh

> /usr/bin/node
> NodeJS status = 0
> Installer is using your system NodeJS.
> 
>  -> Root directory is /home/v/.config
>  -> Creating a directory at /home/v/.config/google-chrome/NativeMessagingHosts
>  -> Chrome Browser is supported
>  -> Creating a directory at /home/v/.config/chromium/NativeMessagingHosts
>  -> Chromium Browser is supported
>  -> Creating a directory at /home/v/.config/vivaldi/NativeMessagingHosts
>  -> Vivaldi Browser is supported
>  -> Creating a directory at /home/v/.config/BraveSoftware/Brave-Browser/NativeMessagingHosts
>  -> Brave Browser is supported
>  -> Creating a directory at /home/v/.config/microsoftedge/NativeMessagingHosts
>  -> Microsoft Edge Browser is supported
>  -> Creating a directory at /home/v/.mozilla/native-messaging-hosts
>  -> Firefox Browser is supported
>  -> Creating a directory at /home/v/.waterfox/native-messaging-hosts
>  -> Waterfox Browser is supported
>  -> Creating a directory at /home/v/.tor-browser/app/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts
>  -> Tor Browser is supported
>  -> Creating a directory at /home/v/.thunderbird/native-messaging-hosts
>  -> Thunderbird Email Client is supported
>  -> Creating a directory at /home/v/.config/com.add0n.node
>  => Native Host is installed in /home/v/.config/com.add0n.node
> 
> 
> >>> host is ready <<<

Seems ok right?

But in Native Integration page, I click Check Connection, a pop-up appears saying
"Cannot find the native client. Follow the 3 steps to install the native client"

Console messages (2 per click):
> AbortError: Native application start canceled by user [NativeMessaging.jsm:185](resource://gre/modules/NativeMessaging.jsm)
>     _doInitPortal resource://gre/modules/NativeMessaging.jsm:185
>     InterpretGeneratorResume self-hosted:1822
>     AsyncFunctionThrow self-hosted:814

and

> NS_ERROR_ILLEGAL_VALUE: Component returned failure code: 0x80070057 (NS_ERROR_ILLEGAL_VALUE) [nsINativeMessagingPortal.closeSession]

Canceled by user? This is a fresh Ubuntu install. Worked perfectly yesterday with Arch. I've installed all programs that are required 
I tried the same with Chrome, Waterfox, what have you but I couldn't get it to work with anything, so I don't think it's a Firefox issue..

Anyone have any idea what this error means?
Cheers

**Comment by SigmFreud** [comment]
> Seems ok right?

No. As already [started here](https://github.com/andy-portmen/native-client/issues/108) directory was needed:
 ```
-> Creating a directory at /home/user/snap/firefox/common/com.add0n.node
 => Native Host is installed in /home/user/snap/firefox/common/com.add0n.node
...
 -> Creating a directory at /home/user/snap/firefox/common/.mozilla/native-messaging-hosts
-> Firefox Browser (snap version) is supported 
```
And ffmpeg installed to:
`/home/user/snap/firefox/common/ffmpeg`

But this method stopped working already in ubuntu 22.04.02:
![error](https://user-images.githubusercontent.com/105107554/236622147-c2034cc0-e48b-42e8-81c0-81d2627db1ac.png)

Ok. I tried to specify the path manually. 
Open this file:
`~/.config/com.add0n.node/run.sh `
And change "./node" to "/home/user/snap/firefox/common/com.add0n.node/node". As a result:
```
#!/usr/bin/env bash
/home/user/snap/firefox/common/com.add0n.node/node host.js
```
But now there is another error that I do not know how to fix:
NS_ERROR_ILLEGAL_VALUE: Component returned failure code: 0x80070057 (NS_ERROR_ILLEGAL_VALUE) [nsINativeMessagingPortal.closeSession]
![Screenshot from 2023-05-06 14-59-00](https://user-images.githubusercontent.com/105107554/236622751-0c2d5227-1d9a-49ac-b211-3abeae44ca22.png)

**Comment by jandede** [comment]
@SigmFreud thanks for clarification.

I've concluded that this is most likely an issue with Snap. I've been out of the scene for a long time so I hardly knew what it even is! However I found this (unrelated) FF addon has issues with specifically the Snap version of FF
https://github.com/browserpass/browserpass-native/issues/112

I then installed non-snap version of Firefox [113.0 (64-bit)], and this add-on works once again. Wohoo!

Also, a lot of similiar add-ons started working again (like "Open with"), didn't work with that Snap version... So most likely it's snap that is at fault here

**Comment by jandede** [comment]
Yep,
Apparently it is being worked on, but this problem has been around for a long time:
https://forum.snapcraft.io/t/native-messaging-support-in-strictly-confined-browser-snaps/26849

**Comment by Mohammad-gif** [comment]
(,\_\°+minhaj enter#'|)(’i'···'p300pharm,-íì-,)_m`-s|.·data'-^-’°–°–°\sumon20[Gulf]\-.|_.|dip&soikat_ò”swift
